src/pages/Calendar/Calendar-Flow.vue

```diff
@@ -0,0 +1,248 @@
+<script>
+import FlowName from '@/pages/Calendar/FlowName'
+import { mapGetters } from 'vuex'
+import { formatTime } from '@/mixins/formatTimeMixin'
+
+export default {
+  components: {
+    FlowName
+  },
+  filters: {},
+  mixins: [formatTime],
+  props: {
+    projectId: {
+      required: false,
+      type: String,
+      default: () => null
+    },
+    timeInterval: {
+      required: true,
+      type: Number
+    },
+    date: {
+      required: true,
+      type: String
+    }
+  },
+  data() {
+    return {
+      skip: true,
+      loadingKey: 0,
+      gettingRuns: false,
+      flowRunEvents: [],
+      selectedEvent: null,
+      selectedOpen: false,
+      selectedElement: null
+    }
+  },
+  computed: {
+    ...mapGetters('api', ['isCloud']),
+    ...mapGetters('tenant', ['tenant']),
+    ...mapGetters('user', ['timezone']),
+    intervalCount() {
+      return (60 / this.timeInterval) * 24
+    },
+    end() {
+      return this.addDay(this.date, 1)
+    },
+    flowIds() {
+      const ids = this.flowRuns?.map(flowRun => flowRun.flow_id)
+      return ids
+    },
+    overlay() {
+      return this.loadingKey > 0 || this.gettingRuns
+    }
+  },
+  watch: {
+    async date() {
+      this.gettingRuns = true
+      this.flowRunEvents = await this.flowRunEventsMethod()
+      this.gettingRuns = false
+    }
+  },
+  async created() {
+    this.flowRunEvents = await this.flowRunEventsMethod()
+  },
+  methods: {
+    async flowRunEventsMethod() {
+      this.gettingRuns = true
+      const timeGroups = this.timeGroups()
+      let event = []
+      timeGroups.forEach(async (timeGroup, i) => {
+        const finished = await this.$apollo.query({
```

Comment:
Same note about these methods as above - can store the variables in a single variable and re-use them and also use `Promise.all` to let them run in parallel.